### PR TITLE
Feature/add react router

### PR DESCRIPTION
**Pull request by Akin909**
I added react router to the project and split the main component out into a home component which is now rendered by react router.

There was also an issue with the methods on the filter component which is new syntax which I believe the brunch setup isn't configured to support so I changed the syntax to the slightly older syntax to get it working (I'm more familiar with webpack so a bit unclear how to otherwise fix it).

Other stylistic changes were due to my linter/auto formatter but if they're an issue I can change those back.

### CODE REVIEW

**Comment by cafreeman** [comment]
@Akin909 Can you take a look at the updated version of the react SPA branch? The babel issues have been fixed, so a lot of your syntax changes (like adding `.bind(this)` back in and getting rid of the instance methods) should probably be reverted.

File: apps/web/assets/js/components/Home.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  handleFilterChange(filters) {
+    this.setState(prevState => ({
+      filters: filters,
+    }));
+  }
+
+  render() {
+    return (
+      <div>
+        <div>
```
Comment (cafreeman) [inline apps/web/assets/js/components/Home.jsx]: This looks like it's probably an extra `div`

Comment (Akin909) [reply apps/web/assets/js/components/Home.jsx]: Will do didn't want to mess with the brunch config as I've never used it before didnt seem like an ideal solution to revert syntax ill revert those changes and get rid of the extra div

**Comment by doomspork** [comment]
Thank you @Akin909 👍

**Comment by doomspork** [comment]
Hey @Akin909, do you have a twitter handle? 😁

**Comment by Akin909** [comment]
Hey @doomspork thanks for merging that I can have a look at anymore react issues open on the repo if there's more stuff that needs doing? 
PS. my twitter handle is @Akin_So

**Comment by doomspork** [comment]
@Akin909 definitely!  I'm busy most of today with work but I will spend some time tonight trying to hash out some more features in the Issues.  I'd love to hear your thoughts on ideas for the project as well!  Anything you think we could improve?

**Comment by Akin909** [comment]
@doomspork sorry for the late response, I could have a look at the filtering of the results #65 , Regarding the project in general I know there was a discussion previously regarding using brunch but I think in the context of working with react I think webpack is a more popular choice so in terms of adding functionality down the line I think it'd be easier to find react ecosystem contributors, compatible plugins, advice or tips re setup using webpack (caveat of course being thats this one guys opinion), otherwise I love the idea curious about elixir myself.

**Comment by doomspork** [comment]
@Akin909 myself or @ybur-yug (or many of the other collaborators) would be happy to help you get involved in the Elixir side of the code if you're interested!  Don't want to have you feeling stuck doing React stuff 😀

**Pull request merged**